src/server/saved_objects/service/lib/repository.js

```diff
@@ -476,12 +477,21 @@ export class SavedObjectsRepository {
 
   async _callCluster(method, params) {
     try {
+      this._assertMigrated();
       return await this._unwrappedCallCluster(method, params);
     } catch (err) {
       throw decorateEsError(err);
     }
   }
 
+  _assertMigrated() {
+    if (!this._migrator.isMigrated) {
+      const error = new Error(`The saved object index ${this._index} is being migrated.`);
```

Comment:
The SavedObjects Repository generally throws errors defined here: https://github.com/elastic/kibana/blob/master/src/server/saved_objects/service/lib/errors.js

Reply:
Good to know. I'll move the error definition there.